src/managers/config.py

```diff
@@ -139,61 +140,26 @@ def current_properties(self) -> list[str]:
         """The current configuration properties set to zoo.cfg."""
         return self.workload.read(self.workload.paths.properties)
 
-    @property
-    def current_env(self) -> list[str]:
-        """The current /etc/environment variables."""
-        return self.workload.read(path="/etc/environment")
-
-    @property
-    def static_properties(self) -> list[str]:
-        """Build the zoo.cfg content, without dynamic options.
-
-        Returns:
-            List of static properties to compared to current zoo.cfg
-        """
-        return self.build_static_properties(self.dashboard_properties)
-
     def set_dashboard_properties(self) -> None:
         """Writes built config file."""
         self.workload.write(
-            content="\n".join(self.dashboard_properties),
+            content=yaml.dump(self.dashboard_properties),
             path=self.workload.paths.properties,
         )
 
-    @staticmethod
-    def build_static_properties(properties: list[str]) -> list[str]:
-        """Removes dynamic config options from list of properties.
-
-        Args:
-            properties: the properties to make static
-
-        Returns:
-            List of static properties
-        """
-        return [
-            prop
-            for prop in properties
-            if ("clientPort" not in prop and "secureClientPort" not in prop)
-        ]
+    def load_dashboard_properties(self) -> dict[str, Any]:
+        """Reads built config file."""
+        return yaml.safe_load(
+            "\n".join(
+                self.workload.read(
+                    path=self.workload.paths.properties,
+                )
+            )
+        )
 
     def config_changed(self) -> bool:
         """Compares expected vs actual config that would require a restart to apply."""
-        server_properties = self.build_static_properties(self.current_properties)
-        config_properties = self.static_properties
-
-        properties_changed = set(server_properties) ^ set(config_properties)
-
-        if not properties_changed:
+        if self.load_dashboard_properties() == self.dashboard_properties:
             return False
-
-        if properties_changed:
-            logger.info(
```

Comment:
Properties have sensitive information, we should not log it. We do not do these diffs for OpenSearch either.